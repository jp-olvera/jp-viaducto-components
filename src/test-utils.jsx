import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
