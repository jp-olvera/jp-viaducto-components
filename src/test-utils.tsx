import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
