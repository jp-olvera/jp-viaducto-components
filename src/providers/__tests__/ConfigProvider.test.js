/* eslint-env jest */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '../../test-utils';
import { ConfigProvider, ConfigContext } from '..';

describe('<ConfigProvider/>', () => {
  test('should create snapshot of ConfigContext', () => {
    const updateConfig = jest.fn();
    const { container } = render(
      <ConfigContext.Provider value={{ updateConfig }}>
        <div>AAAAA</div>
      </ConfigContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('should create snapshot ConfigProvider', () => {
    const setState = jest
      .fn()
      .mockImplementation(() => ({ spacing: { hola: '1rem' } }));
    const useStateMock = jest.spyOn(React, 'useState');
    useStateMock.mockImplementation((init) => [
      {
        ...init,
        ...{ spacing: { hola: 1 }, toasterPlacement: 'centerFailed' },
      },
      setState,
    ]);
    const { container } = render(
      <ConfigProvider>
        <div>AAAAA</div>
      </ConfigProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
