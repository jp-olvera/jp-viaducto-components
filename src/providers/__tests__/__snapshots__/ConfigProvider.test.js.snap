// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigProvider/> should create snapshot ConfigProvider 1`] = `
<div>
  <div>
    AAAAA
  </div>
</div>
`;

exports[`<ConfigProvider/> should create snapshot of ConfigContext 1`] = `
<div>
  <div>
    AAAAA
  </div>
</div>
`;
